eox_nelp/user_profile/api/v1/views.py

```diff
@@ -0,0 +1,79 @@
+"""Generic views for NELP user's profile application.
+
+Function-views:
+  - generate_otp: generate and send via SMS the OTP related a user. Saved in cache.
+  - validate_otp: Compare and check if the proposed OTP match the User OTP saved in cache.
+    If match, updates the profile phone_number.
+
+"""
+import logging
+
+from django.core.cache import cache
+from django.db import transaction
+from django.http import HttpResponseForbidden, JsonResponse
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from rest_framework import status
+from rest_framework.decorators import api_view, authentication_classes, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from eox_nelp.edxapp_wrapper.user_api import accounts, errors
+
+logger = logging.getLogger(__name__)
+
+
+@api_view(["POST"])
+@authentication_classes((
+    SessionAuthenticationAllowInactiveUser,
+))
+@permission_classes((IsAuthenticated,))
+def update_user_data(request):
+    """ View to update user's fields.
+    ## Usage
+
+    ### **POST** /eox-nelp/api/user-profile/v1/update-user-data/
+
+    request example data:
+    ``` json
+    {
+        "phone_number": 3213123123,
+        "one_time_password": "234fasds"
+    }
+    ```
+    **POST Response Values**
+    ``` json
+    {
+        "message": "User's fields has been updated successfully"
+    }
+    ```
+    """
+    user_phone_number = request.data.get("phone_number", None)
+    proposed_user_otp = request.data.get("one_time_password", None)
+
+    if not user_phone_number or not proposed_user_otp:
+        return JsonResponse(
+            data={"detail": "missing phone_number or one_time_password in data."},
+            status=status.HTTP_400_BAD_REQUEST,
+        )
+
+    user_otp_key = f"{request.user.username}-{user_phone_number}"
+    logger.info("validating otp for %s*****", user_otp_key[:-5])
+
+    if not proposed_user_otp == cache.get(user_otp_key):
+        return HttpResponseForbidden(reason="Forbidden - wrong code")
+
+    try:
+        with transaction.atomic():
```

Comment:
Why do you need to be atomic?

Reply:
Why do you select the atomic logic in context manager and not in decorator way to use?
https://docs.djangoproject.com/en/3.2/topics/db/transactions/#controlling-transactions-explicitly

Reply:
You can find the answer of the first question in this [link ](https://github.com/nelc/edx-platform/blob/open-release/palm.nelp/openedx/core/djangoapps/user_api/accounts/api.py#L130)

same thing different implementation, if you prefer one of them please expose your arguments

Reply:
I am ok, I only want to know if there was a particular reason. Anyway, no problem.